### ISSUE TITLE
Assertion `d % (RESULT_STEPS * BLOCKDIM_Z) == 0' failed

### ISSUE THREAD

**Issue by ok37** [lsb-riken/CUBIC-informatics#2]
Hello,

I was attempting run HDoG_gpu.py using the test data and kept running into this error. Any insight on what might be causing this?

GPU:0	images:100
GPU:0	paramfile: ./param/param_gpu00.json
Executing : ./build/HDoG3D_main ./param/param_gpu00.json
HDoG3D_main: /home/ok37/repos/CUBIC-informatics/src/3D/NormalizeFilter.cu:519: void minmaxLayers3D(short unsigned int*, short unsigned int*, int, int, int, int) [with int BLOCKDIM_X = 8; int BLOCKDIM_Y = 8; int BLOCKDIM_Z = 16; int RESULT_STEPS = 2; int HALO_STEPS = 2; bool is_min = true]: Assertion `d % (RESULT_STEPS * BLOCKDIM_Z) == 0' failed.
[*] finished

**Comment by s-horiguchi** [comment]
Hi ok37,

I suppose you change the `depth` parameter in the parameter file to reduce GPU memory consumption.
The CUDA program is fine-tuned for maximum performance, so you need to adjust not only the parameter file but also the constants defined in the source code in this case.

In the assertion `d % (RESULT_STEPS * BLOCKDIM_Z) == 0`, `d` corresponds to `depth` parameter. So it means `depth` parameter should be divisible by `RESULT_STEPS*BLOCKDIM_Z=2*16=32`. 

For example, if you want to set `depth` to `10`, I recommend setting `BLOCKDIM_Z` to `5` which is defined in `src/3D/NormalizeFilter.cu` line 40 as `L_LAY_BLOCKDIM_Z`.

I hope it works!

**Comment by ok37** [comment]
Thanks for replying! In my implementation, I set `depth = 16` and `BLOCKDIM_Z 8` in `src/3D/NormalizeFilter.cu`. I also had to make the same change for `LAYERS_BLOCKDIM_Z` in `src/3D/DoG3DFilter.cu`. This led me to get some memory errors in `RegionalFeatures.cu` where `Assertion temp_storage_bytes < cub_tmp_bytes' failed.`, even if I increased `cub_tmp_size_factor`. I ended up commenting these assertions out and all the other steps ran without any issues. The results looked similar to what was is in the notebook
